keyless/pepper/service/src/jwk.rs

```diff
@@ -4,18 +4,20 @@
 use crate::{metrics::JWK_FETCH_SECONDS, Issuer, KeyID};
 use anyhow::{anyhow, Result};
 use aptos_logger::warn;
+use aptos_types::jwks::rsa::RSA_JWK;
 use dashmap::DashMap;
-use jsonwebtoken::{jwk::JwkSet, DecodingKey};
+use jsonwebtoken::DecodingKey;
 use once_cell::sync::Lazy;
+use serde_json::Value;
 use std::{sync::Arc, time::Duration};
 use tokio::time::Instant;
 
 /// The JWK in-mem cache.
-pub static DECODING_KEY_CACHE: Lazy<DashMap<Issuer, DashMap<KeyID, Arc<DecodingKey>>>> =
+pub static DECODING_KEY_CACHE: Lazy<DashMap<Issuer, DashMap<KeyID, Arc<RSA_JWK>>>> =
```

Comment:
wonder what made you have to refactor this?

Reply:
This is to match the Prover code that does is (we need to dedupe the code there).

Rex notes - 
// TODO: this is a duplicate of the jwk fetching in the pepper service, with changes b/c the
// DecodingKey type that the pepper service uses is too opaque to use here. We should unify.

And the point is that turning is into decoding key prematurely loses information we need and we cannot recover.  I don't think there is a isomorphic function between DecodingKey <-> RSA_JWK.

The kty is tough to recover.  But we need it as it is used to compute public hash.

Reply:
it would be better if it's `Arc<JWK>>` instead of `Arc<RSA_JWK>` but it's fine rn...